allennlp/training/trainer.py

```diff
@@ -948,13 +961,11 @@ def from_params(cls,
         patience = params.pop_int("patience", None)
         validation_metric = params.pop("validation_metric", "-loss")
         num_epochs = params.pop_int("num_epochs", 20)
-        cuda_device = params.pop_int("cuda_device", -1)
+        cuda_device = params.pop( "cuda_device")
```

Comment:
This parameter still needs to have a default - this is causing a lot of the CI to break.

Reply:
Done - sorry I missed this to begin with